mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -124,6 +124,7 @@ def __init__(self, *,  # pylint: disable=too-many-locals
             random_design=random_design,
             config_selector=config_selector,
             overwrite=True,
+            logging_level=False,  # Use the existing logger
```

Comment:
Since it's supposed to be possible to use `mlos_core` independently of `mlos_bench` can we:
- in `mlos_core` extend the constructor params to accept a `logging_level`
  - if it's `None` (default), then issue a warning that `SMAC` will be interacting with the logger somehow
- in `mlos_bench` we set that value for them by default

Thoughts?

Reply:
By the way, that SMAC `logging_level` parameter is actually quite a mess! I've looked at the source code, and it can actually be of one of *four* different types:
* `False`: do not configure the logger (i.e., use the default / already configured logger)
* `None`: reconfigure the logger with SMAC defaults (use INFO level, write to stdout only, reset the output format, etc.)
* `int`: (e.g., `20`): reconfigure the logger with SMAC defaults, but set the logging level to the given threshold
* `str`: A path to the YAML configuration file (!) that contains logging parameters.

Now imagine every optimizer that we use in `mlos_core` being so creative about their logging! We'll end up implementing our own adapters to configure each optimizer's logging, etc., etc. I would much prefer each optimizer rely on some preconfigured logging instead. Every other way is just not composable

Reply:
Ack.  You've convinced me.